src/commons/utils/alerts.py

```diff
@@ -0,0 +1,163 @@
+#
+# -*- coding: utf-8 -*-
+# !/usr/bin/python
+"""
+Module for generating email
+"""
+import json
+import logging
+import os
+import smtplib
+import threading
+import time
+from email.mime.application import MIMEApplication
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.utils import formatdate, COMMASPACE, make_msgid
+
+from config import CORIO_CFG
+from src.commons import commands
+from src.commons.constants import ROOT
+from src.commons.degrade_cluster import get_logical_node
+from src.commons.utils.corio_utils import get_report_file_path
+
+LOGGER = logging.getLogger(ROOT)
+
+
+# pylint: disable=too-few-public-methods
+class Mail:
+    """Module to send mail"""
+
+    def __init__(self, sender, receiver):
+        """
+        Init method
+        :param sender: email address of sender
+        :param receiver: email address of receiver
+        """
+        self.mail_host = os.getenv("EMAIL_HOST")
+        self.port = int(os.getenv("EMAIL_PORT"))
+        self.sender = sender
+        self.receiver = receiver
+
+    def send_mail(self, message):
+        """
+        Function to send mail using smtp server
+        :param message: Email message
+        """
+        LOGGER.info("Sending mail with Subject %s:", message['Subject'])
+        with smtplib.SMTP(self.mail_host, self.port) as server:
+            server.sendmail(self.sender, self.receiver.split(','), message.as_string())
+
+
+class MailNotification(threading.Thread):
+    """This class contains common utility methods for Mail Notification."""
+
+    def __init__(self, sender, receiver, tp_id, report_path):
+        """
+        Init method:
+        :param sender: sender of mail
+        :param receiver: receiver of mail
+        :param tp_id : Test Plan ID to be sent in subject.
+        :param report_path: Report file path.
+        """
+        super().__init__()
+        self.event_fail = threading.Event()
+        self.mail_obj = Mail(sender=sender, receiver=receiver)
+        self.sender = sender
+        self.receiver = receiver
+        self.tp_id = tp_id
+        self.health_obj = get_logical_node()
+        self.report_path = report_path
+        self.message_id = None
+
+    def prepare_email(self, execution_status) -> MIMEMultipart:
+        """
+        Prepare email message with format and attachment
+        :param execution_status: Execution status. In Progress/Fail
+        :return: Formatted MIME message
+        """
+        hctl_status = json.dumps(self.health_obj.get_hctl_status()[1], indent=4)
+        result, pod_status = self.health_obj.execute_command(commands.CMD_POD_STATUS)
+        status = f"Corio TestPlan {str(self.tp_id or '')} is {execution_status} " \
+                 f"on {self.health_obj.host}"
+        subject = status
+        body = f"<h3>{status}.</h2>\n" \
+               f"<h3>PFA hctl cluster status, pod status & execution status.</h3>\n"
```

Comment:
can we add active sessions in mail?

Reply:
This will be part of report and not email. Reports are attached to emails.